### PR TITLE
[3.9] Respect Laravel 9's single word name mutators

### PR DESCRIPTION
**Pull request by RosemaryOrchard**
This is an updated version of PR [2394](https://github.com/jenssegers/laravel-mongodb/pull/2394) fixing the formatting and adding unit tests.

Laravel 9 introduced single word mutator functions (e.g. if you have a property `columns`, then your mutator is `public function columns()`, and currently this library causes it to return null.

### CODE REVIEW

**Comment by RosemaryOrchard** [comment]
I believe StyleCI is misconfigured contrary to the `.php-cs-fixer.dist.php`? All the tests pass when run on my own fork

File: src/Eloquent/Model.php [hunk 1]
```diff
@@ -155,7 +155,11 @@ public function getAttribute($key)
         }
 
         // This checks for embedded relation support.
-        if (method_exists($this, $key) && ! method_exists(self::class, $key)) {
+        if (
+            method_exists($this, $key)
+            && ! method_exists(self::class, $key)
+            && ! $this->hasAttributeGetMutator($key)
```
Comment (Smolevich) [inline src/Eloquent/Model.php:161]: Add additional test cases, please

**Comment by Smolevich** [comment]
> I believe StyleCI is misconfigured contrary to the `.php-cs-fixer.dist.php`? All the tests pass when run on my own fork

Yes, but only maintainer @jenssegers can change settings in StyleCI

File: src/Eloquent/Model.php (hunk 1 above)
Comment (RosemaryOrchard) [reply src/Eloquent/Model.php:161]: I've added a test in the Model/User. If there is anywhere else appropriate or relevant please let me know :)

**Review by Smolevich — APPROVED** [review]

**Review by divine — APPROVED** [review]

**Pull request merged**